File: spring-cloud-gateway/gateway-starter/src/main/java/com/livk/dynamic/annotation/EnableDynamicGateway.java
package com.livk.dynamic.annotation;

import com.livk.commons.spring.context.AutoImport;

import java.lang.annotation.*;

/**
 * <p>
 * EnableDynamicGateway
 * </p>
 *
 * @author livk
 * @date 2021/12/28
 */
@Target({ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
@Documented
@Inherited
@AutoImport
public @interface EnableDynamicGateway {

}

File: spring-cloud-gateway/gateway-starter/src/main/kotlin/com/livk/dynamic/annotation/EnableDynamicGateway.kt
package com.livk.dynamic.annotation

import com.livk.commons.annotation.AutoImport
import java.lang.annotation.Inherited

/**

 * @author livk
 */
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
@Inherited
@AutoImport
annotation class EnableDynamicGateway
